### PR TITLE
Enable TypeScript strict mode + fix up issues

### PR DESCRIPTION
**Pull request by akx**
Originally from #14.

This PR:

* enables `strict` and fixes up some problems uncovered by it
* moves the `ang()` inner function to a free function
* fixes up some unnecessary variable reassignment/aliasing

### CODE REVIEW

File: src/svg-to-paths.ts [hunk 1]
```diff
@@ -137,22 +146,25 @@ export function flattenSVG(svg: SVGElement, options: Partial<Options> = {}): Lin
...
+    let closePoint: Point | null = null
     for (const cmd of pathData) {
       if (cmd.type === 'M') {
         cur = xf(cmd.values)
         closePoint = cur
         paths.push({
           points: [cur],
-          stroke: getStroke(shape),
-          groupId: getGroupId(shape)
+          stroke: getStroke(shape) ?? undefined,
```
Comment (nornagon) [inline src/svg-to-paths.ts]: `?? undefined` makes no sense to me... return undefined unless the value is undefined?

File: src/svg-to-paths.ts [hunk 2]
```diff
@@ -137,22 +146,25 @@ export function flattenSVG(svg: SVGElement, options: Partial<Options> = {}): Lin
...
+          groupId: getGroupId(shape) ?? undefined
         });
       } else if (cmd.type === 'L') {
         cur = xf(cmd.values)
         paths[paths.length-1].points.push(cur)
       } else if (cmd.type === 'C') {
         const [x1, y1, x2, y2, x3, y3] = cmd.values
+        if(cur === null) {
```
Comment (nornagon) [inline src/svg-to-paths.ts]:
```suggestion
        if (cur === null) {
```

File: src/svg-to-paths.ts [hunk 3]
```diff
@@ -137,22 +146,25 @@ export function flattenSVG(svg: SVGElement, options: Partial<Options> = {}): Lin
...
+          throw new Error(`C ${cmd} encountered without current point`)
```
Comment (nornagon) [inline src/svg-to-paths.ts]: `cmd` is an object here right? Won't this show `C [object Object] ...`?

File: src/svg-to-paths.ts [hunk 4]
```diff
@@ -163,12 +175,11 @@ export function flattenSVG(svg: SVGElement, options: Partial<Options> = {}): Lin
...
       } else if (cmd.type === 'A') {
-        const [rx_, ry_, xAxisRotation, largeArc, sweep, x, y] = cmd.values
-        const phi = xAxisRotation
-        const fS = sweep
-        const fA = largeArc
-        const {cos, sin, atan2, sqrt, sign, acos, abs, ceil} = Math
-
+        const [rx_, ry_, phi, fA, fS, x, y] = cmd.values
+        const {cos, sin, sqrt, acos, abs, ceil} = Math
+        if(cur === null) {
```
Comment (nornagon) [inline src/svg-to-paths.ts]:
```suggestion
        if (cur === null) {
```

File: src/svg-to-paths.ts [hunk 5]
```diff
@@ -232,6 +235,9 @@ export function flattenSVG(svg: SVGElement, options: Partial<Options> = {}): Lin
         }
         cur = point(x, y)
       } else if (cmd.type === 'Z') {
+        if(cur === null) {
```
Comment (nornagon) [inline src/svg-to-paths.ts]: same here

**Review by nornagon — CHANGES_REQUESTED** [review]

File: src/svg-to-paths.ts (hunk 1 above)
Comment (akx) [reply src/svg-to-paths.ts]: `??` also defers to the right-hand side of the operator if the left side is `null`. 

`getStroke()` returns `string | null`, but `Line` defines `stroke` as `stroke?: string;`, so we need to turn that `null` into `undefined` (or change `Line`'s type).

File: src/svg-to-paths.ts (hunk 3 above)
Comment (akx) [reply src/svg-to-paths.ts]: Good catch! Changing that to `cmd.values`.

Just needed to add some `throw` here so the type of  `cur` gets narrowed to non-null, didn't think about the message too closely.

Comment (akx) [reply src/svg-to-paths.ts]: On that note, I'll give

https://github.com/nornagon/flatten-svg/blob/6d6cde877e028381def7cc488c2609208ebc50bc/src/svg-to-paths.ts#L239

the same treatment 😁

File: src/svg-to-paths.ts (hunk 1 above)
Comment (nornagon) [reply src/svg-to-paths.ts]: We should either change `getStroke` to return `string | undefined` or change `Line` to have the type `string | null`.

**Comment by nornagon** [comment]
LGTM except for the `undefined` / `null` nitpick!

File: src/svg-to-paths.ts (hunk 1 above)
Comment (akx) [reply src/svg-to-paths.ts]: @nornagon I changed `getStroke` and `getGroupId` to return `undefined` instead of `null`. 👍

**Review by nornagon — APPROVED** [review]

**Pull request merged**